### DISCUSSION TITLE
setting reporting_time for "manuSpecificTuya" cluster

### DISCUSSION THREAD

**Discussion by DarioKTS** [Koenkk/zigbee2mqtt#16856] [Devices]
Hello,

in my network i have 7 tuya battery operated sensors:
1x [air quality sensor](https://www.zigbee2mqtt.io/devices/TS0601_air_quality_sensor.html)  
6x [soil sensors](https://www.zigbee2mqtt.io/devices/TS0601_soil.html)

My controller is a cc2531 ( i read, less capable), I am running the Z-Stack_Home_1.2 default firmware published on 15th nov 2021.

I'd like to slow down the reporting times, the soil sensors are reporting every 10 seconds(unneeded), the air quality one is literally flooding the controller with transmissions every second or so. On a few occasions the cc2531 "hung", as in Z2M started failing to connect to the adapter , typing lsusb -v  , i was getting for dev/ttyACM0 a few error lines at the bottom with "resource temproarily unavailable". a full power cycle solves the issue, but only temporarily.
ATM these 7 sensors are the only nodes present.

While i might consider purchasing a more capable controller, i still will have the follwoing issue:

In [this guide](https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-request-device-configure-reporting)  is described how to set reporting time. From my understanding  the expected syntax needs to have an "attribute" field, so much so that from the Z2M frontend it's even impossible to send the reporting time setting request without it, but manuSpecificTuya doesn't have an attribute filed, as it is a collection of tuya datapoints+ values.
I've however tried it directly via mqtt, to no avail.

I have also read [this ](https://github.com/Koenkk/zigbee2mqtt/issues/3771)older issue, from prior to when it was possible to do it via mqtt, and i guess all the  devices using manuSpecificTuya might need to be treated this way.

please help,

thx in advance

**Comment by DarioKTS** [comment]
after some fiddling in tuya.js with the "ts0601_soil" ( i know i'm not supposed to, but i'm eager to get things working), inserting 

```
            configure: async (device) => {
                const endpoint = device.getEndpoint(1);
            await configureReporting.temperature(endpoint,{minimumReportInterval:1000 , maximumReportInterval: 36000});
            await configureReporting.soil_moisuture(endpoint,{minimumReportInterval:1000 , maximumReportInterval: 36000});
            await configureReporting.battery(endpoint,{minimumReportInterval:1000 , maximumReportInterval: 36000});
            await configureReporting.batteryState(endpoint,{minimumReportInterval:3000 , maximumReportInterval: 36000});
            },
```
achieved exactly nothing XD...
(the humongous numbers were to make sure, for testing purposes)


Going back to : please help!

**Reply by kkossev** [reply]
Unfortunately, there is no known way to change the hardcoded reporting interval for these two Tuya devices.

**Reply by DarioKTS** [reply]
Ty for letting me know

**Comment by King-anderson** [comment]
I have a Zemismart 1ph powermeter. It makes the Zigbee network crowded. When I use 8 pcs of powermeters, Z2M will be very slow. I use Tuya hub. Smartlife app notifies an update request for the powermeter. after updating it over the air, report interval is added to the device. but I don't know how set the reporting for each dp. Fortunately, after updating the preset report configuration is remained and zigbee network is not crowded.
